offchain protocol DIP-1 support

xli opened the pull request.

LuZhang-Lou commented on src/diem/offchain/types/schema.py:
```diff
@@ -0,0 +1,124 @@
+
+
+class NationalIdObject(BaseSchema[data_types.NationalIdObject]):
+    id_value = fields.Str(allow_none=True)
```
id_value can't be none
https://github.com/diem/lip/blob/master/lips/lip-1.mdx#nationalidobject

LuZhang-Lou commented on src/diem/offchain/types/data_types.py:
```diff
@@ -0,0 +1,217 @@
+
+
+class ErrorCode:
+    reject_kyc_data = "rejected"
+    no_kyc_needed = "no-kyc-needed"
```
I think these two are `abort_code` rather than `error_code`

LuZhang-Lou reviewed (commented):
I agree with you that wallet.py seems too complex so that understanding what each function call does may not be trivial for users. 
It may make sense to integrate this with one example (e.g. RW?) to get an initial sense. Let's do it?

LuZhang-Lou commented:
and I like how you leverage marmarshmallow to define data structs nice and clean

xli commented:
> and I like how you leverage marmarshmallow to define data structs nice and clean

switched to define scheme with the data class, and removed marshmallow, because the marshmallow validation error has no details of reason.

xli commented:
> I agree with you that wallet.py seems too complex so that understanding what each function call does may not be trivial for users.
> It may make sense to integrate this with one example (e.g. RW?) to get an initial sense. Let's do it?

I will create a PR for LRW Wed.

sunmilee commented on src/diem/offchain/__init__.py:
```diff
@@ -0,0 +1,56 @@
+import typing
+
+
+def travel_rule_metadata(payment: PaymentObject, hrp: str) -> typing.Tuple[bytes, bytes]:
```
why is this defined here?

xli replied:
convert payment: PaymentObject into travel rule metadata, 

offchain API is higher level comparing with DIP-4 implementation, so this func calls to txnmetadata.travel_rule to create data

xli replied:
this is removed and merged into PaymentCommand related functions.
